src/main/java/net/dv8tion/jda/core/requests/restaction/ChannelOrderAction.java

```diff
@@ -0,0 +1,61 @@
+/*
+ *     Copyright 2015-2016 Austin Keener & Michael Ritter
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.dv8tion.jda.core.requests.restaction;
+
+import net.dv8tion.jda.core.entities.*;
+import net.dv8tion.jda.core.requests.Route;
+import org.apache.http.util.Args;
+import org.json.JSONArray;
+import org.json.JSONObject;
+
+import java.util.Collection;
+
+public class ChannelOrderAction<T extends Channel> extends OrderAction<T, ChannelOrderAction>
+{
+    protected final Guild guild;
```

Comment:
I think we should also add `getChannelType()` that would either return `TEXT` or `VOICE` depending on the boolean value passed to the constructor. In addition to be more future-proof (channel categories) we should maybe just pass a `ChannelType` directly instead of using a boolean to determine the order type.